flink-architecture-tests/flink-architecture-tests-base/src/main/java/org/apache/flink/architecture/common/Predicates.java

```diff
@@ -157,6 +166,35 @@ public static <T> DescribedPredicate<T> exactlyOneOf(
                 t -> Arrays.stream(other).map(dp -> dp.test(t)).reduce(false, Boolean::logicalXor));
     }
 
+    /**
+     * Extracts the class name from the given fully qualified class name.
+     *
+     * <p>Example:
+     *
+     * <pre>
+     *     getClassFromFqName("com.example.MyClass");  // Returns: "MyClass"
+     * </pre>
+     */
+    public static String getClassSimpleNameFromFqName(String fqClassName) {
```

Comment:
One thing to be careful of is what kind of fqcn you are using.

Canonical names can for example contain a `$` before the simple name if you are dealing with an inner class. Or its the other way around that getName returns it; this always trips me up.

Reply:
Thanks for pointing this out. I changed the approach to handle both cases and added test coverage for this method:
https://github.com/apache/flink/pull/23491/commits/fbea7c9382bd9d3f6a7d80c81e05e8491a137612